exec/csnex/Support.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace csnex
+{
+    public class Support
+    {
+        static public List<string> Paths;
+
+        public Support(string source_path)
+        {
+            Paths = new List<string>();
+
+            if (source_path != null && source_path.Length > 0) {
+                // Check the Neon executable path, first.
+                Paths.Add(string.Format("{0}{1}", source_path, Path.DirectorySeparatorChar));
+            }
+            // Then, check current directory;
+            Paths.Add(string.Format(".{0}", Path.DirectorySeparatorChar));
+
+            // Get paths from %NEONPATH% environment variable, if there is one.
+            string ne = Environment.GetEnvironmentVariable("NEONPATH");
+            if (ne != null) {
+                string[] evpaths = ne.Split(Path.PathSeparator);
+                foreach (string s in evpaths) {
+                    // Don't bother pushing empty paths.
+                    if (s != null && s.Length > 0) {
+                        Paths.Add(s);
+                    }
+                }
+            }
+            // Next, check for .neonpath, and process any paths that might exist in it.
+            try {
+                StreamReader sr = new StreamReader(".neonpath");
+                string line = null;
+                do {
+                    line = sr.ReadLine();
+                    // Don't bother pushing empty paths.
+                    if (line != null && line.Length > 0) {
```

Comment:
You've got confusing logic here and a double check for null. Check once for null and use break to get out of the loop.